glm-ocr-pro/ocr_local.py

```
@@ -0,0 +1,152 @@
+    }
+
+    req = request.Request(
+        f"{OLLAMA_BASE_URL}/api/chat",
+        method="POST",
+        headers={"Content-Type": "application/json"},
+        data=json.dumps(payload).encode("utf-8"),
+    )
+
+    try:
+        # We manually iterate over response lines for streaming
+        with request.urlopen(req, timeout=timeout_seconds) as resp:
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)